lib/ASRouter.jsm

```diff
@@ -125,7 +123,15 @@ class _ASRouter {
     this.initialized = false;
     this.messageChannel = null;
     this._storage = null;
+    this._prefs = null;
     this._resetInitialization();
+    this._updateProviderEndpointUrl = this._updateProviderEndpointUrl.bind(this);
+    this._addASRouterPrefListener();
```

Comment:
You're calling this twice, once in the constructor and once in `init()`; i'm assuming you just meant to call it in init?